flow/scripts/global_route.tcl

```diff
@@ -23,7 +23,7 @@ proc global_route_helper { } {
 
   if { $result != 0 } {
     if {
-      [!$::env(GENERATE_ARTIFACTS_ON_FAILURE) || \
+      [expr !$::env(GENERATE_ARTIFACTS_ON_FAILURE) || \
```

Comment:
Do we need the expr here?  I think the problem is the outer [] are not needed.

Reply:
@luarss Aha!

Now we know what happened.

Fixing warnings is dangerous  :-) Removing the "expr" fixed the warning in this unusual .tcl syntax, but introduced a bug.

```
tclfmt 0.4.2
flow/scripts/global_route.tcl:26:7: unnecessary command substitution within expression [redundant-expr]
Error: Process completed with exit code 1.
```